crates/biome_js_analyze/src/analyzers/nursery/no_useless_ternary.rs

```diff
@@ -97,3 +204,66 @@ fn is_boolean_literal(expression: &AnyJsExpression) -> bool {
     }
     false
 }
+
+fn is_boolean_expression(expression: &AnyJsExpression) -> Option<bool> {
+    match expression.syntax().kind() {
+        JsSyntaxKind::JS_BINARY_EXPRESSION => {
+            let operator = expression
+                .as_js_binary_expression()?
+                .operator_token()
+                .ok()?;
+            let operator = operator.text_trimmed();
+            if BOOLEAN_OPERATORS.contains(&operator) {
+                return Some(true);
+            }
+        }
+        JsSyntaxKind::JS_UNARY_EXPRESSION => {
+            let operator = expression.as_js_unary_expression()?.operator_token().ok()?;
+            let operator = operator.text_trimmed();
+            if operator == "!" {
+                return Some(true);
+            }
```

Comment:
Don't check the content of the operator, use the utilities we have in the syntax crate.